opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -0,0 +1,150 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from logging import getLogger
+from math import inf
+
+from opentelemetry._metrics.instrument import _Monotonic
+from opentelemetry.util._time import _time_ns
+
+_logger = getLogger(__name__)
+
+
+class Aggregation(ABC):
+    @property
+    def value(self):
+        return self._value  # pylint: disable=no-member
+
+    @abstractmethod
+    def aggregate(self, value):
+        pass
+
+    @abstractmethod
+    def make_point_and_reset(self):
+        pass
+
+
+class NoneAggregation(Aggregation):
+    """
+    This aggregation drops all instrument measurements.
+    """
+
+    def __init__(self, instrument):
+        self._value = None
+
+    def aggregate(self, value):
+        pass
+
+    def make_point_and_reset(self):
+        pass
+
+
+class SumAggregation(Aggregation):
+    """
+    This aggregation collects data for the SDK sum metric point.
+    """
+
+    def __init__(self, instrument):
+        self._value = 0
+
+    def aggregate(self, value):
+        self._value = self._value + value
+
+    def make_point_and_reset(self):
+        pass
+
+
+class LastValueAggregation(Aggregation):
+
+    """
+    This aggregation collects data for the SDK sum metric point.
+    """
+
+    def __init__(self, instrument):
+        self._value = None
+        self._timestamp = _time_ns()
+
+    def aggregate(self, value):
+        self._value = value
+        self._timestamp = _time_ns()
+
+    def make_point_and_reset(self):
+        pass
+
+
+class ExplicitBucketHistogramAggregation(Aggregation):
+
+    """
+    This aggregation collects data for the SDK sum metric point.
+    """
+
+    def __init__(
+        self,
+        instrument,
+        *args,
+        boundaries=(0, 5, 10, 25, 50, 75, 100, 250, 500, 1000, inf),
```

Comment:
Why do we have inf here? The [default in the spec doesn't have it](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#:~:text=%5B%200%2C%205%2C%2010%2C%2025%2C%2050%2C%2075%2C%20100%2C%20250%2C%20500%2C%201000%20%5D) because it is implicitly the upper bound for the last bucket bound ([see proto](https://github.com/open-telemetry/opentelemetry-proto/blob/8672494217bfc858e2a82a4e8c623d4a5530473a/opentelemetry/proto/metrics/v1/metrics.proto#L561))

Reply:
Removed